### PR TITLE
Unresolve filter expression before printing

### LINKED ISSUE #6655: Dates in explain are not human readable
e.g
```
    └─ ScanFilterProject[table = hive:sqlperformance:query_10_dates_100000_rows, grouped = false, 
filterPredicate = ((("@date@52QIVV94J9UG605IGOB1J45N16Q5GQ6J6I4M25GR4PSJLVN4VMCGP03MMVE0VIR14H0IRIH6N1CP8SS618IQDH4501E8FFFP54506G94EMPHPLOR42US53BJMQ4KIGVKQOGP1MGMUNJBH5RUHR4UB5BJQIR9F232AS10C079567IDKSSHPISKREE2LUV29034G31H4O="("date") >= DATE '2020-07-01') 
AND ("@date@52QIVV94J9UG605IGOB1J45N16Q5GQ6J6I4M25GR4PSJLVN4VMCGP03MMVE0VIR14H0IRIH6N1CP8SS618IQDH4501E8FFFP54506G94EMPHPLOR42US53BJMQ4KIGVKQOGP1MGMUNJBH5RUHR4UB5BJQIR9F232AS10C079567IDKSSHPISKREE2LUV29034G31H4O="("date") <= DATE '2020-12-31')) 
AND ("@date@52QIVV94J9UG605IGOB1J45N16Q5GQ6J6I4M25GR4PSJLVN4VMCGP03MMVE0VIR14H0IRIH6N1CP8SS618IQDH4501E8FFFP54506G94EMPHPLOR42US53BJMQ4KIGVKQOGP1MGMUNJBH5RUHR4UB5BJQIR9F232AS10C079567IDKSSHPISKREE2LUV29034G31H4O="("date") BETWEEN DATE '2020-07-01' AND DATE '2020-12-31'))]
           Layout: [h310_0:varchar, date:varchar, $path_1:varchar, $file_size_2:bigint, $file_modified_time_3:timestamp(3) with time zone, $partition_4:varchar, date_5:date]
           Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: 0B}/{rows: ? (?), cpu: ?, memory: 0B, network: 0B}/{rows: ? (?), cpu: ?, memory: 0B, network: 0B}
           CPU: 140.00ms (0.00%), Scheduled: 2.33s (0.03%), Output: 100000 rows (2.35MB)
           Input avg.: 10000.00 rows, Input std.dev.: 0.00%
           date_5 := date("date")
           date := date:string:PARTITION_KEY
               :: [[2020-07-01], [2020-07-02], [2020-07-03], [2020-07-04], [2020-07-05], [2020-07-06], [2020-07-07], [2020-07-08], [2020-07-09], [2020-07-10]]
           $partition_4 := $partition:string:SYNTHESIZED
           h310_0 := h310:string:REGULAR
           $path_1 := $path:string:SYNTHESIZED
           $file_modified_time_3 := $file_modified_time:timestamp:SYNTHESIZED
           $file_size_2 := $file_size:bigint:SYNTHESIZED
           Input: 100000 rows (1.94MB), Filtered: 0.00%
```
CC @dain

### PR DESCRIPTION
**Pull request by dain**
fixes printing of:
```
create table d as  select '2020-01-01' ds;

explain select count(*) from d where date(ds) < date '2020-01-02'
```

Fixes #6655

### CODE REVIEW

**Review by findepi — APPROVED** [review]
Can we you please add a test for `EXPLAIN` that it's output (the single `String`) contains predicate in fully human readable form?

**Comment by findepi** [comment]
This fixes https://github.com/trinodb/trino/issues/6655, right?

**Review by sopel39 — APPROVED** [review]
some test would be nice. Maybe some test similar to: `io.trino.tests.TestDistributedEngineOnlyQueries#testExplainAnalyze?

File: testing/trino-tests/src/test/java/io/trino/tests/TestDistributedEngineOnlyQueries.java [hunk 1]
```diff
@@ -137,6 +138,14 @@ public void testRowTypeWithReservedKeyword()
                 "VALUES 1");
     }
 
+    @Test
+    public void testExplain()
+    {
+        assertExplain(
+                "explain select name from nation where abs(nationkey) = 22",
```
Comment (findepi) [inline testing/trino-tests/src/test/java/io/trino/tests/TestDistributedEngineOnlyQueries.java:145]: uppercase keywords: EXPLAIN SELECT ..

also, as reported in https://github.com/trinodb/trino/issues/6655, can date be any special?

**Pull request merged**